### PR TITLE
SGD8-546: Webform bugfixes

### PR DESCRIPTION
**Pull request by hstandaert**
## Description

* Added header to webforms
* Display description for fieldsets
* make sure the aria-describedby attribute links to the corre… …
* Display select at full width of the form-column
* display city and postal code next to each other
* Display date/time fields as stacked
* Don't allow italic text in error messages.
* Add title for webform confirmation page

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the gent_base CHANGELOG accordingly.

### CODE REVIEW

File: theme.inc [hunk 1]
```diff
@@ -167,27 +167,40 @@ function _gent_base_add_webform_step_title(array &$form) {
...
-    $build = [];
+    _gent_base_add_webform_header($step, $step['#title']);
+  }
+}
+
+/**
+ * Adds the step title to the step for webforms with multiple pages.
+ *
+ * @param array $form
+ *   Render array for the form.
```
Comment (lennartvava) [inline theme.inc]:
```suggestion
 *   Render array for the form.
 * @param string $title
 *   The title of the form step.
```

File: source/sass/11-base/forms/_forms.scss [hunk 2]
```diff
@@ -31,3 +31,15 @@ textarea {
   @include theme('color', 'color-tertiary--lighten-1', 'form-description-color');
   font-size: .6rem;
 }
+
```
Comment (lennartvava) [inline source/sass/11-base/forms/_forms.scss:34]: Dubbele new line?

**Review by meirege — APPROVED** [review]

**Review by lennartvava — APPROVED** [review]

File: includes/theme.inc [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+/**
+ * Helper function to get the file data of a referenced File entity.
+ *
+ * @param mixed $variables
+ *   The variables array to work with.
+ * @param string $field
+ *   The field to get the referenced file file size from.
+ *
+ * @return array|null
+ *   The file size of the document in bytes.
```
Comment (lennartvava) [inline includes/theme.inc]: Return value is een array met meer data dan document size.
```suggestion
 *   Array containing the file_size, file_uri and file_type.
```

File: templates/contrib/webform/webform.html.twig [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+ *
+ * @see template_preprocess_webform()
+ * @see _webform_form_after_build()
+ *
+ * @ingroup themeable
+ */
+#}
+
+<form{{ attributes }}>
+
```
Comment (lennartvava) [inline templates/contrib/webform/webform.html.twig]:
```suggestion
```

File: includes/theme.inc [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+function _gent_base_get_entity_reference_file_data(&$variables, $field) {
+  $data = [];
+
+  /** @var \Drupal\paragraphs\Entity\Paragraph $paragraph */
+  $paragraph = $variables['elements']['#paragraph'];
+  if (!$paragraph->hasField($field)) {
+    return;
+  }
+  /* @var $entity \Drupal\Core\Field\Plugin\Field\FieldType\EntityReferenceItem */
+  $entity = $paragraph->get($field)->first();
```
Comment (lennartvava) [inline includes/theme.inc:62]: Gaat altijd maar over 1 value? Nooit multi value?

File: includes/form.inc [hunk 6]
```diff
@@ -2,148 +2,31 @@
...
 
-  foreach ($array as &$field) {
+  foreach ($children as $key) {
+    $field = $elements[$key];
     if (!is_array($field) || !isset($field['#type'])) {
       continue;
     }
 
-    if (!in_array($field['#type'], $fields_types)) {
+    if (!isset($field['#value'])) {
```
Comment (lennartvava) [inline includes/form.inc:30]:
```suggestion
    // Only input elements have a #value property.
    if (!isset($field['#value'])) {
```

File: includes/theme.inc (hunk 5 above)
Comment (hstandaert) [reply includes/theme.inc:62]: Wordt enkel gebruikt om `field_primary_action_document` en `field_secondary_action_document` op te halen, wat nooit meer dan 1 file zou mogen zijn (want dat is niet ondersteund)

**Pull request merged**